design-documents/graph-ql/framework/attributes-metadata.md

```diff
@@ -0,0 +1,58 @@
+# Problem statement
+
+Storefront applications built on top of Magento GraphQL web APIs need to support the following scenarios:
+ - Render advanced catalog search form, which can be configured by the merchant
+ - Display only a subset of all available product attributes in catalog listing based on the configuration
+ - Render product compare page, which should only display attributes specified by the merchant
+ 
+All these scenarios require specific EAV attribute metadata to be accessible from the storefront application.
+
+Additionally, there should be a way to retrieve metadata for all storefront custom attributes in the system. This information should be cached on the server and on the client. 
+
+# Proposed solution
+
+Relaxing signature of existing `customAttributeMetadata` query by making its `attriubtes` argument optional will allow to fetch all storefront attributes metadata.
```

Comment:
Could you please add current representation of `customAttributeMetadata` to the document?

Reply:
Done